src/com/jwetherell/algorithms/strings/StringFunctions.java

```diff
@@ -14,14 +14,26 @@ public class StringFunctions {
 
     public static final String reverseWithStringConcat(String string) {
         String output = new String();
+        
+        if(string==null || string.equals("")){
+            return "";
```

Comment:
Added handling for null or empty string inputs. Return empty string in such cases